server/api/plextv.ts

```diff
@@ -253,6 +287,81 @@ class PlexTvAPI {
     )) as UsersResponse;
     return parsedXml;
   }
+
+  public async getWatchlist({
+    offset = 0,
+    size = 20,
+  }: { offset?: number; size?: number } = {}): Promise<{
+    offset: number;
+    size: number;
+    totalSize: number;
+    items: PlexWatchlistItem[];
+  }> {
+    try {
+      const response = await this.axios.get<WatchlistResponse>(
+        '/library/sections/watchlist/all',
+        {
+          params: {
+            'X-Plex-Container-Start': offset,
+            'X-Plex-Container-Size': size,
+          },
+          baseURL: 'https://metadata.provider.plex.tv',
+        }
+      );
+
+      const watchlistDetails = await Promise.all(
+        response.data.MediaContainer.Metadata.map(async (watchlistItem) => {
```

Comment:
Ah, found one more thing... this can be `undefined` if the user's watchlist is empty.

```
2022-08-21T06:26:31.346Z [error][Plex.TV Metadata API]: Failed to retrieve watchlist items{"errorMessage":"Cannot read properties of undefined (reading 'map')"}
```

Reply:
Good catch!